core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabase.java

```diff
@@ -60,8 +59,7 @@ public JdbcDatabase(DatabaseConfig databaseConfig) {
             databaseConfig.getMetadataCacheExpirationTimeSecs());
 
     OperationChecker operationChecker = new OperationChecker(databaseConfig, tableMetadataManager);
-    QueryBuilder queryBuilder = new QueryBuilder(rdbEngine);
-    jdbcService = new JdbcService(tableMetadataManager, operationChecker, queryBuilder);
+    jdbcService = new JdbcService(tableMetadataManager, operationChecker, rdbEngine);
```

Comment:
Push down `RdbEngineStrategy` since we don't have to create it again at the deeper level (i.e., `ConditionMutator`).